### PR TITLE
FirstPersonRenderer -> HeldItemRenderer

### PR DESCRIPTION
**Pull request by Sollace**
Just made a PR dealing with the `ItemRenderer` and `FirstPersonRenderer` classes making the change I suggested in discord earlier today.

I also made sure to go through and update/properly map all the methods and parameters in both files.

### CODE REVIEW

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
+CLASS net/minecraft/class_759 net/minecraft/client/render/Item/HeldItemRenderer
```
Comment (ChloeDawn) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]: Capital `I` in package name

**Review by ChloeDawn — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping (hunk 1 above)
Comment (Sollace) [reply mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]: oh

**Review by liach — CHANGES_REQUESTED** [review]
Please don't call a matrix stack a matrix. call it matrices at least.

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 2]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 4 player
+		ARG 5 light
 	METHOD method_3215 resetEquipProgress (Lnet/minecraft/class_1268;)V
+		ARG 1 hand
 	METHOD method_3216 renderArm (Lnet/minecraft/class_4587;Lnet/minecraft/class_4597;ILnet/minecraft/class_1306;)V
+		ARG 1 matrix
+		ARG 2 provider
+		ARG 3 light
+		ARG 4 arm
+	METHOD method_3217 applywingOffset (Lnet/minecraft/class_4587;Lnet/minecraft/class_1306;F)V
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]: hmm, what is applywing?

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 3]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
+	METHOD method_23179 renderItem (Lnet/minecraft/class_1799;Lnet/minecraft/class_809$class_811;ZLnet/minecraft/class_4587;Lnet/minecraft/class_4597;IILnet/minecraft/class_1087;)V
+		ARG 1 stack
+		ARG 2 transformation
+		ARG 3 isLeft
+		ARG 4 matrix
+		ARG 5 provider
+		ARG 6 light
+		ARG 7 overlay
+		ARG 8 model
+	METHOD method_23180 renderBasedItemQuads (Lnet/minecraft/class_4587;Lnet/minecraft/class_4588;Ljava/util/List;Lnet/minecraft/class_1799;II)V
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]: based? what does that mean, baked?

**Comment by Sollace** [comment]
> 
> 
> Please don't call a matrix stack a matrix. call it matrices at least.

Well I've called it `stack` in other places, but can't do that when you have a `ItemStack`.

I think `matrix` is a good enough qualifier to describe it, because most of the time you're just using it as that.

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping (hunk 3 above)
Comment (Sollace) [reply mappings/net/minecraft/client/render/item/ItemRenderer.mapping]: Typo

**Comment by Sollace** [comment]
Who's decision was it to call it a `MatrixStack` anyway?

**Comment by liach** [comment]
Mind updating to pre4?

**Comment by Sollace** [comment]
@liach Updated and base branch changed to 1.15-pre5

**Comment by Sollace** [comment]
@liach Updated and base branch changed to 1.15-pre6

**Review by modmuss50 — APPROVED** [review]

File: mappings/net/minecraft/client/render/GameRenderer.mapping [hunk 4]
```diff
@@ -45,6 +45,9 @@ CLASS net/minecraft/class_757 net/minecraft/client/render/GameRenderer
 		ARG 2 scaledHeight
 		ARG 3 tickDelta
 	METHOD method_3172 renderHand (Lnet/minecraft/class_4587;Lnet/minecraft/class_4184;F)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/GameRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/entity/EntityRenderer.mapping [hunk 5]
```diff
@@ -29,6 +29,6 @@ CLASS net/minecraft/class_897 net/minecraft/client/render/entity/EntityRenderer
 		ARG 2 yaw
 		ARG 3 tickDelta
 		ARG 4 matrices
-		ARG 5 vertexConsumers
+		ARG 5 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/entity/EntityRenderer.mapping]:
```suggestion
		ARG 5 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping (hunk 2 above)
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
	METHOD method_3217 applySwingOffset (Lnet/minecraft/class_4587;Lnet/minecraft/class_1306;F)V
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 6]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+	METHOD method_22976 renderItem (FLnet/minecraft/class_4587;Lnet/minecraft/class_4597$class_4598;Lnet/minecraft/class_746;I)V
+		ARG 1 tickDelta
+		ARG 2 matrix
+		ARG 3 provider
+		ARG 4 player
+		ARG 5 light
 	METHOD method_3215 resetEquipProgress (Lnet/minecraft/class_1268;)V
+		ARG 1 hand
 	METHOD method_3216 renderArm (Lnet/minecraft/class_4587;Lnet/minecraft/class_4597;ILnet/minecraft/class_1306;)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 7]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 2 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 2 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 8]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
 	FIELD field_4053 prevEquipProgressMainHand F
 	METHOD <init> (Lnet/minecraft/class_310;)V
 		ARG 1 client
+	METHOD method_22976 renderItem (FLnet/minecraft/class_4587;Lnet/minecraft/class_4597$class_4598;Lnet/minecraft/class_746;I)V
+		ARG 1 tickDelta
+		ARG 2 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 2 matrices
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 9]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 3 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 3 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 10]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 5 light
 	METHOD method_3215 resetEquipProgress (Lnet/minecraft/class_1268;)V
+		ARG 1 hand
 	METHOD method_3216 renderArm (Lnet/minecraft/class_4587;Lnet/minecraft/class_4597;ILnet/minecraft/class_1306;)V
+		ARG 1 matrix
+		ARG 2 provider
+		ARG 3 light
+		ARG 4 arm
+	METHOD method_3217 applywingOffset (Lnet/minecraft/class_4587;Lnet/minecraft/class_1306;F)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 11]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 2 arm
+		ARG 3 swingProgress
 	METHOD method_3218 applyEatOrDrinkTransformation (Lnet/minecraft/class_4587;FLnet/minecraft/class_1306;Lnet/minecraft/class_1799;)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 12]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 2 tickDelta
+		ARG 3 arm
+		ARG 4 stack
 	METHOD method_3219 renderArmHoldingItem (Lnet/minecraft/class_4587;Lnet/minecraft/class_4597;IFFLnet/minecraft/class_1306;)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 13]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 2 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 2 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 14]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 3 light
+		ARG 4 equipProgres
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 4 equipProgress
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 15]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 5 swingProgres
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 5 swingProgress
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 16]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 6 arm
 	METHOD method_3220 updateHeldItems ()V
 	METHOD method_3222 renderMapInOneHand (Lnet/minecraft/class_4587;Lnet/minecraft/class_4597;IFLnet/minecraft/class_1306;FLnet/minecraft/class_1799;)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 17]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 2 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 2 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 18]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 3 light
+		ARG 4 equipProgress
+		ARG 5 arm
+		ARG 6 swingProgres
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 6 swingProgress
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 19]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 7 stack
 	METHOD method_3223 renderFirstPersonMap (Lnet/minecraft/class_4587;Lnet/minecraft/class_4597;ILnet/minecraft/class_1799;)V
-	METHOD method_3224 applyHandOffset (Lnet/minecraft/class_4587;Lnet/minecraft/class_1306;F)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 20]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 2 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 2 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 21]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 3 swingProgress
+		ARG 4 stack
+	METHOD method_3224 applyEquipOffset (Lnet/minecraft/class_4587;Lnet/minecraft/class_1306;F)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 22]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 1 tickDelta
 	METHOD method_3228 renderFirstPersonItem (Lnet/minecraft/class_742;FFLnet/minecraft/class_1268;FLnet/minecraft/class_1799;FLnet/minecraft/class_4587;Lnet/minecraft/class_4597;I)V
 		ARG 1 player
 		ARG 2 tickDelta
 		ARG 3 pitch
 		ARG 4 hand
+		ARG 5 swingProgress
 		ARG 6 item
 		ARG 7 equipProgress
+		ARG 8 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 8 matrices
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 23]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 9 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 9 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 24]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 10 light
 	METHOD method_3231 renderMapInBothHands (Lnet/minecraft/class_4587;Lnet/minecraft/class_4597;IFFF)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 25]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 2 provier
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 2 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 26]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 3 light
+		ARG 4 pitch
+		ARG 5 equipProgress
+		ARG 6 swingProgress
 	METHOD method_3233 renderItem (Lnet/minecraft/class_1309;Lnet/minecraft/class_1799;Lnet/minecraft/class_809$class_811;ZLnet/minecraft/class_4587;Lnet/minecraft/class_4597;I)V
+		ARG 1 entity
+		ARG 2 stack
+		ARG 3 transformation
+		ARG 4 isLeft
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 4 leftHanded
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 27]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 1 matrix
+		ARG 2 provier
+		ARG 3 light
+		ARG 4 pitch
+		ARG 5 equipProgress
+		ARG 6 swingProgress
 	METHOD method_3233 renderItem (Lnet/minecraft/class_1309;Lnet/minecraft/class_1799;Lnet/minecraft/class_809$class_811;ZLnet/minecraft/class_4587;Lnet/minecraft/class_4597;I)V
+		ARG 1 entity
+		ARG 2 stack
+		ARG 3 transformation
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 3 transformationType
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 28]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 4 isLeft
+		ARG 5 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 5 matrices
```

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping [hunk 29]
```diff
@@ -10,21 +10,78 @@ CLASS net/minecraft/class_759 net/minecraft/client/render/FirstPersonRenderer
...
+		ARG 6 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
		ARG 6 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 30]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
 	FIELD field_4733 colorMap Lnet/minecraft/class_325;
 	METHOD <init> (Lnet/minecraft/class_1060;Lnet/minecraft/class_1092;Lnet/minecraft/class_325;)V
+		ARG 1 manager
+		ARG 2 bakery
 		ARG 3 colorMap
+	METHOD method_23177 renderItem (Lnet/minecraft/class_1309;Lnet/minecraft/class_1799;Lnet/minecraft/class_809$class_811;ZLnet/minecraft/class_4587;Lnet/minecraft/class_4597;Lnet/minecraft/class_1937;II)V
+		ARG 1 entity
+		ARG 2 item
+		ARG 3 transformation
+		ARG 4 isLeft
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 4 leftHanded
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 31]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
 	FIELD field_4732 models Lnet/minecraft/class_763;
 	FIELD field_4733 colorMap Lnet/minecraft/class_325;
 	METHOD <init> (Lnet/minecraft/class_1060;Lnet/minecraft/class_1092;Lnet/minecraft/class_325;)V
+		ARG 1 manager
+		ARG 2 bakery
 		ARG 3 colorMap
+	METHOD method_23177 renderItem (Lnet/minecraft/class_1309;Lnet/minecraft/class_1799;Lnet/minecraft/class_809$class_811;ZLnet/minecraft/class_4587;Lnet/minecraft/class_4597;Lnet/minecraft/class_1937;II)V
+		ARG 1 entity
+		ARG 2 item
+		ARG 3 transformation
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 3 transformationType
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 32]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
+		ARG 4 isLeft
+		ARG 5 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 5 matrices
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 33]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
+		ARG 6 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 6 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 34]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
+		ARG 7 world
+		ARG 8 light
+		ARG 9 overlay
+	METHOD method_23178 renderItem (Lnet/minecraft/class_1799;Lnet/minecraft/class_809$class_811;IILnet/minecraft/class_4587;Lnet/minecraft/class_4597;)V
+		ARG 1 stack
+		ARG 2 transformation
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 2 transformationType
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 35]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
+		ARG 3 light
+		ARG 4 overlay
+		ARG 5 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 5 matrices
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 36]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
+		ARG 6 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 6 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 37]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
+	METHOD method_23179 renderItem (Lnet/minecraft/class_1799;Lnet/minecraft/class_809$class_811;ZLnet/minecraft/class_4587;Lnet/minecraft/class_4597;IILnet/minecraft/class_1087;)V
+		ARG 1 stack
+		ARG 2 transformation
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 2 transformationType
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 38]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
+		ARG 3 isLeft
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 3 leftHanded
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 39]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
+		ARG 4 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 4 matrices
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 40]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
+		ARG 5 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 5 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping (hunk 3 above)
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
	METHOD method_23180 renderBakedItemQuads (Lnet/minecraft/class_4587;Lnet/minecraft/class_4588;Ljava/util/List;Lnet/minecraft/class_1799;II)V
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 41]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
+		ARG 6 light
+		ARG 7 overlay
+		ARG 8 model
+	METHOD method_23180 renderBasedItemQuads (Lnet/minecraft/class_4587;Lnet/minecraft/class_4588;Ljava/util/List;Lnet/minecraft/class_1799;II)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping [hunk 42]
```diff
@@ -6,12 +6,56 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
...
 		ARG 0 vertexConsumers
 		ARG 1 layer
+		ARG 2 solid
 		ARG 3 glint
+	METHOD method_23182 renderBakedItemModel (Lnet/minecraft/class_1087;Lnet/minecraft/class_1799;IILnet/minecraft/class_4587;Lnet/minecraft/class_4588;)V
+		ARG 1 model
+		ARG 2 stack
+		ARG 3 light
+		ARG 4 overlay
+		ARG 5 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/item/ItemRenderer.mapping]:
```suggestion
		ARG 5 matrices
```

File: mappings/net/minecraft/client/render/model/json/Transformation.mapping [hunk 43]
```diff
@@ -9,6 +9,9 @@ CLASS net/minecraft/class_804 net/minecraft/client/render/model/json/Transformat
 		ARG 3 scale
 	METHOD equals (Ljava/lang/Object;)Z
 		ARG 1 o
+	METHOD method_23075 apply (ZLnet/minecraft/class_4587;)V
+		ARG 1 isLeft
```
Comment (liach) [inline mappings/net/minecraft/client/render/model/json/Transformation.mapping]:
```suggestion
		ARG 1 leftHanded
```

File: mappings/net/minecraft/client/render/model/json/Transformation.mapping [hunk 44]
```diff
@@ -9,6 +9,9 @@ CLASS net/minecraft/class_804 net/minecraft/client/render/model/json/Transformat
...
+		ARG 2 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/model/json/Transformation.mapping]:
```suggestion
		ARG 2 matrices
```

File: mappings/net/minecraft/client/render/model/json/Transformation.mapping [hunk 45]
```diff
@@ -20,4 +23,4 @@ CLASS net/minecraft/class_804 net/minecraft/client/render/model/json/Transformat
 		METHOD method_3493 parseVector3f (Lcom/google/gson/JsonObject;Ljava/lang/String;Lnet/minecraft/class_1160;)Lnet/minecraft/class_1160;
 			ARG 1 json
 			ARG 2 key
-			ARG 3 default_
+			ARG 3 def
```
Comment (liach) [inline mappings/net/minecraft/client/render/model/json/Transformation.mapping]:
```suggestion
			ARG 3 fallback
```

**Review by liach — CHANGES_REQUESTED** [review]
:rage: I am seriously triggered by how you marked my pointing out of your typos as resolved while you did not fix them at all!

File: mappings/net/minecraft/client/render/entity/EntityRenderer.mapping (hunk 5 above)
Comment (Sollace) [reply mappings/net/minecraft/client/render/entity/EntityRenderer.mapping]: * `provider`

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping (hunk 31 above)
Comment (Sollace) [reply mappings/net/minecraft/client/render/item/ItemRenderer.mapping]: What value does including `type` have here? It's a transformation.

**Comment by Sollace** [comment]
> 
> 
> 😡 I am seriously triggered by how you marked my pointing out of your typos as resolved while you did not fix them at all!

Sorry, I legitimately thought I'd fixed them. I even remember pushing the commit.

File: mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping [hunk 46]
```diff
@@ -1,5 +1,20 @@
 CLASS net/minecraft/class_1007 net/minecraft/client/render/entity/PlayerEntityRenderer
+	METHOD method_23205 renderArm (Lnet/minecraft/class_4587;Lnet/minecraft/class_4597;ILnet/minecraft/class_742;Lnet/minecraft/class_630;Lnet/minecraft/class_630;)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping [hunk 47]
```diff
@@ -1,5 +1,20 @@
...
+		ARG 2 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping]:
```suggestion
		ARG 2 vertexConsumers
```

File: mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping [hunk 48]
```diff
@@ -1,5 +1,20 @@
...
+		ARG 3 light
+		ARG 4 player
+		ARG 5 arm
+		ARG 6 sleeve
 	METHOD method_4210 getArmPose (Lnet/minecraft/class_742;Lnet/minecraft/class_1799;Lnet/minecraft/class_1799;Lnet/minecraft/class_1268;)Lnet/minecraft/class_572$class_573;
 	METHOD method_4218 setModelPose (Lnet/minecraft/class_742;)V
 	METHOD method_4220 renderRightArm (Lnet/minecraft/class_4587;Lnet/minecraft/class_4597;ILnet/minecraft/class_742;)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping [hunk 49]
```diff
@@ -1,5 +1,20 @@
...
+		ARG 2 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping]:
```suggestion
		ARG 2 vertexConsumers
```

File: mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping [hunk 50]
```diff
@@ -1,5 +1,20 @@
...
+		ARG 3 light
+		ARG 4 player
 	METHOD method_4221 renderLeftArm (Lnet/minecraft/class_4587;Lnet/minecraft/class_4597;ILnet/minecraft/class_742;)V
+		ARG 1 matrix
```
Comment (liach) [inline mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping]:
```suggestion
		ARG 1 matrices
```

File: mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping [hunk 51]
```diff
@@ -1,5 +1,20 @@
...
+		ARG 2 provider
```
Comment (liach) [inline mappings/net/minecraft/client/render/entity/PlayerEntityRenderer.mapping]:
```suggestion
		ARG 2 vertexConsumers
```

File: mappings/net/minecraft/client/render/item/ItemRenderer.mapping (hunk 31 above)
Comment (liach) [reply mappings/net/minecraft/client/render/item/ItemRenderer.mapping]: The param type is `net/minecraft/client/render/model/json/ModelTransformation$Type`

**Review by liach — COMMENTED** [review]
Sorry, missed a few other places.

File: mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping (hunk 1 above)
Comment (liach) [inline mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]:
```suggestion
CLASS net/minecraft/class_759 net/minecraft/client/render/item/HeldItemRenderer
```
Seems the name in git is correct.

Comment (Sollace) [reply mappings/net/minecraft/client/render/item/HeldItemRenderer.mapping]: Yeah, I don't know what is going on here. I fixed this before as well.

**Comment by Sollace** [comment]
I think this PR is cursed.

**Review by liach — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Review by Runemoro — APPROVED** [review]

**Pull request merged**